cpp/src/fil/fil.cu

```diff
@@ -915,12 +1011,30 @@ void from_treelite(const raft::handle_t& handle, forest_t* pforest,
 void from_treelite(const raft::handle_t& handle, forest_t* pforest,
                    ModelHandle model, const treelite_params_t* tl_params) {
   const tl::Model& model_ref = *(tl::Model*)model;
-  model_ref.Dispatch([&handle, pforest, tl_params](const auto& model_inner) {
+  model_ref.Dispatch([&](const auto& model_inner) {
     // model_inner is of the concrete type tl::ModelImpl<T, L>
     from_treelite(handle, pforest, model_inner, tl_params);
   });
 }
 
+template <typename T, typename L, typename N>
+char* sprintf_shape(const tl::ModelImpl<T, L>& model, storage_type_t storage,
+                    const std::vector<N>& nodes,
+                    const std::vector<int>& trees) {
+  std::stringstream forest_shape = depth_hist_and_max(model);
+  float size_mb = (trees.size() * sizeof(trees.front()) +
+                   nodes.size() * sizeof(nodes.front())) /
+                  1e6;
+  char line[100];
```

Comment:
Same as above, it's better to avoid allocating on-stack buffers and use IO manipulators instead.